perf/otel_dev_exporter.go

```diff
@@ -42,33 +44,43 @@ type OtelDevelopmentExporter struct {
 func (e *OtelDevelopmentExporter) ExportSpans(ctx context.Context, spans []sdktrace.ReadOnlySpan) error {
 	e.mu.Lock()
 	defer e.mu.Unlock()
-	if e.shutdown {
-		return nil
+	if err := ctx.Err(); err != nil || e.shutdown {
+		return err
 	}
+	spanInfo := lo.Map(spans, func(span sdktrace.ReadOnlySpan, _ int) *myOtelSpanInfo {
+		return spanInfoFromROSpan(span)
+	})
+	e.exportSpans(spanInfo)
+	return nil
+}
 
-	for _, sd := range spans {
-		length := (sd.EndTime().UnixNano() - sd.StartTime().UnixNano()) / (1000 * 1000)
-		myinfo := fmt.Sprintf("%s %d ms", sd.Name(), length)
-
-		for _, a := range sd.Attributes() {
-			myinfo = myinfo + " " + string(a.Key) + ":" + a.Value.Emit()
-		}
-
-		rawSpanID := sd.SpanContext().SpanID()
-		spanID := hex.EncodeToString(rawSpanID[:])
-		rawParentSpanID := sd.Parent().SpanID()
-		parentSpanID := hex.EncodeToString(rawParentSpanID[:])
+// ExportOTLPSpans displays spans that have already been serialized in OTLP format.
+func (e *OtelDevelopmentExporter) ExportOTLPSpans(ctx context.Context, spans []*tracepb.Span) error {
```

Comment:
Is the implementation of this method in order to satisfy another interface? Should we do one of those `var client _ foo = &OtelDevelopmentExporter{}` if so?

Reply:
No, it's just the version of `ExportSpans` that takes OTLP data. `ExportSpans` is satisfying an interface defined by an Otel package